base/sharedarray.jl

```diff
@@ -512,8 +512,6 @@ function copy!(S::SharedArray, R::SharedArray)
     return S
 end
 
-complex(S1::SharedArray,S2::SharedArray) = convert(SharedArray, complex(S1.s, S2.s))
```

Comment:
`complex.(S1, S2)` would be at the mercy of `similar`s return type here, right? so would not return a `SharedArray` any more unless we were to go back on `similar` returning `SharedArray` ?